metadetect/lsst_metadetect.py

```diff
@@ -1,352 +1,377 @@
-            # if needed
-            step=self['metacal'].get("step", shearpos.DEFAULT_STEP),
-        )
-        res['row_noshear'] = rows_noshear
-        res['col_noshear'] = cols_noshear
-
-    def _add_mfrac(self, res, obs):
-        if np.any(self.mfrac > 0):
-            # we are using the positions with the metacal shear removed for
-            # this.
-            res["mfrac"] = measure_mfrac(
-                mfrac=self.mfrac,
-                x=res["col_noshear"],
-                y=res["row_noshear"],
-                box_sizes=res["box_size"],
-                obs=obs,
-                fwhm=self.get("mfrac_fwhm", None),
-            )
+                psf_image = obs.psf.image
+                stack_psf = KernelPsf(
+                    FixedKernel(
+                        afw_image.ImageD(psf_image.astype(float))
+                    )
+                )
+                exp.setPsf(stack_psf)
+                obs.exposure = exp
+
+                if show:
+                    import descwl_coadd.vis
+                    descwl_coadd.vis.show_image_and_mask(exp)
+                    input('hit a key')
+
+    return odict
+
+
+def get_ormask_and_bmask(mbobs):
+    """
+    get the ormask and bmask, ored from all epochs
+    """
+
+    for band, obslist in enumerate(mbobs):
+        nepoch = len(obslist)
+        assert nepoch == 1, 'expected 1 epoch, got %d' % nepoch
+
+        obs = obslist[0]
+
+        if band == 0:
+            ormask = obs.ormask.copy()
+            bmask = obs.bmask.copy()
         else:
-            res["mfrac"] = 0
+            ormask |= obs.ormask
+            bmask |= obs.bmask
 
-    def _add_original_psf(self, res, shear_str):
-        """
-        TODO add ormask etc.
-        """
+    return ormask, bmask
 
-        res['psfrec_flags'][:] = self.psf_stats['flags']
-        res['psfrec_g'][:, 0] = self.psf_stats['g1']
-        res['psfrec_g'][:, 1] = self.psf_stats['g2']
-        res['psfrec_T'][:] = self.psf_stats['T']
 
-        return res
+def get_mfrac(mbobs):
+    """
+    set the masked fraction image, averaged over all bands
+    """
+    wgts = []
+    mfrac = np.zeros_like(mbobs[0][0].image)
+    for band, obslist in enumerate(mbobs):
+        nepoch = len(obslist)
+        assert nepoch == 1, 'expected 1 epoch, got %d' % nepoch
 
-    def _set_weight(self):
-        wpars = self['weight']
+        obs = obslist[0]
+        wgt = np.median(obs.weight)
+        if hasattr(obs, "mfrac"):
+            mfrac += (obs.mfrac * wgt)
+        wgts.append(wgt)
 
-        T = ngmix.moments.fwhm_to_T(wpars['fwhm'])  # noqa
+    mfrac = mfrac / np.sum(wgts)
+    return mfrac
 
-        # the weight is always centered at 0, 0 or the
-        # center of the coordinate system as defined
-        # by the jacobian
 
-        weight = ngmix.GMixModel(
-            [0.0, 0.0, 0.0, 0.0, T, 1.0],
-            'gauss',
-        )
+def fit_original_psfs(config, mbobs, rng):
```

Comment:
Doesn't this overlap something in fitting.py?

Reply:
the function get_psf_averages assumes a gmix has been stored in the psf obs, so this is a bit different